src/Core/Domain/ApiAccess/Command/EditApiAccessCommand.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+
+class EditApiAccessCommand
+{
+    private ApiAccessId $apiAccessId;
+
+    private ?string $apiClientId = null;
+
+    private ?string $clientName = null;
+
+    private ?bool $enabled = null;
+
+    private ?string $description = null;
+
+    public function __construct(int $apiAccessId)
+    {
+        $this->apiAccessId = new ApiAccessId($apiAccessId);
+    }
+
+    public function getApiAccessId(): ApiAccessId
+    {
+        return $this->apiAccessId;
+    }
+
+    public function setApiAccessId(ApiAccessId $apiAccessId): self
+    {
+        $this->apiAccessId = $apiAccessId;
+
+        return $this;
+    }
```

Comment:
```suggestion
```
The id is a mandatory field forced to be set in the constructor and then it should remain immutable it doesn't need a setter